api/lib/infrastructure/repositories/certification-course-repository.js

```diff
@@ -68,7 +68,7 @@ module.exports = {
   },
 
   async update(certificationCourse) {
-    const certificationCourseData = _adaptModelToDb(certificationCourse);
+    const certificationCourseData = _pickUpdatableProperties(certificationCourse);
```

Comment:
Ca me perturbe que le repo s'occupe de cela. Le choix des champs c'est du metier, pourquoi cela ne serait il pas dans le use case?
(ou au moins renommer le `update`)

Reply:
Ça peut quand même être intéressant de mettre certaines contraintes au plus bas niveau possible, pour garantir une certaine intégrité de base (par exemple, les _foreign keys_ dans la DB ou tout simplement les champs `NOT NULL`). En l'occurrence s'il y a certains champs qui ne devraient absolument jamais être mis à jour (ex. `certification-course.sessionId`…), les bloquer ici ne me paraît pas absurde.
Après ça ne veut pas dire que le code métier doit ignorer le problème, pour moi c'est plutôt une sécurité supplémentaire.

Reply:
C'est ce qu'on a prévu de faire (mettre cette logique dans les usecases) dans la prochaine PR parce que ce qui provoque le bug des certifs désannulées c'est précisément dû au fait que des règles métiers sont dans une méthode de repository appelées par plusieurs use-cases (auxquels on ne devrait pas appliquer les mêmes règles).